R/frgtracker.R

```diff
@@ -18,6 +56,33 @@
 #'register_courses(course_df)
 register_courses <- function(df){
 
+  if (sum(df$course_id %in% mds_courses) != nrow(df)) {
+    stop("I only work on MDS courses! You have at least one is not.")
+  }
+
+  if (sum(df$assessment_id %in% mds_assess) != nrow(df)){
+    stop("I only work on MDS assessments! You have at least one is not.")
+  }
+
+  if (sum(df$weight >= 0) != nrow(df)){
+    stop("You have at least one assessment weight is negative!")
+  }
+
+  w_sum_df <- df %>% group_by(course_id) %>% summarise(w_sum = sum(weight))
+
+  if (sum(w_sum_df$w_sum) != nrow(w_sum_df)) {
+    stop("The sum of all assessment weights should be 1 for individual
+         courses!")
+  }
+
+
+
+  df <- df %>%
+    pivot_wider(names_from = assessment_id, values_from = weight,
+                values_fill = 0)
+  df$course_id <- as.character(df$course_id)
+  df
```

Comment:
normally, people will split so that there is a blank line between the last code line and the object to be returned. So it can be like this:

df$course_id <- as.character(df$course_id)

df
}